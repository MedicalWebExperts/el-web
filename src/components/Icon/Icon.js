import React from 'react';
import { shape, string } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import theme from '../../styles';

const propTypes = {
  styles: shape({}),
  icon: shape({}).isRequired,
  size: string,
  color: string,
  className: string,
};

const defaultProps = {
  styles: {},
  size: 'xs',
  color: theme.colors.textPrimary,
  className: '',
};

const Icon = props => (
  <FontAwesomeIcon
    icon={props.icon}
    size={props.size}
    {...props}
    color={props.color}
    className={props.className}
  />
);

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
