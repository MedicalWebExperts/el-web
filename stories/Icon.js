import React from 'react';
import { storiesOf } from '@storybook/react';
import { faCoffee, faFutbol } from '@fortawesome/free-solid-svg-icons';

import { Icon } from '..';

const customProps = {
  spin: true,
  transform: 'shrink-6 left-4',
  color: 'red',
};

const wrapper = {
  textAlign: 'center',
  marginTop: 100,
};

export default () => {
  storiesOf('Icon', module).add('Icon Default', () => (
    <div style={wrapper}>
      <Icon icon={faCoffee} size="4x" />
    </div>
  ));
  storiesOf('Icon', module).add('Icon Custom Props', () => (
    <div style={wrapper}>
      <Icon icon={faFutbol} size="6x" {...customProps} />
    </div>
  ));
};
