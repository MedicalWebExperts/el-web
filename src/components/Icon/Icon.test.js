import React from 'react';
import renderer from 'react-test-renderer';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

import Icon from './Icon';

const defaultProps = {
  icon: faCoffee,
  size: 'md',
};

const customProps = {
  spin: true,
  transform: 'shrink-6 left-4',
};

describe('Component Icon Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Icon {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom props', () => {
    const tree = renderer
      .create(<Icon {...Object.assign({}, defaultProps, customProps)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
