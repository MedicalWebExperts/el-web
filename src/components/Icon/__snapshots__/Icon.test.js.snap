// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Icon Snapshot using custom props 1`] = `
<div
  className="jsx-994160201 icon"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-coffee fa-w-20 fa-spin fa-lg "
    color="#000000"
    data-icon="coffee"
    data-prefix="fas"
    height="18px"
    role="img"
    style={
      Object {
        "height": "100%",
        "transformOrigin": "0.375em 0.5em",
        "width": "100%",
      }
    }
    styles={Object {}}
    viewBox="0 0 640 512"
    width="18px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      style={Object {}}
      transform="translate(320 256)"
    >
      <g
        style={Object {}}
        transform="translate(-128, 0)  scale(0.625, 0.625)  rotate(0 0 0)"
      >
        <path
          d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
          fill="currentColor"
          style={Object {}}
          transform="translate(-320 -256)"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Component Icon Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-994160201 icon"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-coffee fa-w-20 fa-lg "
    color="#000000"
    data-icon="coffee"
    data-prefix="fas"
    height="18px"
    role="img"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    styles={Object {}}
    viewBox="0 0 640 512"
    width="18px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
